src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1420,20 +1420,27 @@ private object ConvertValue(object value, _SqlMetaData metadata, bool isNull, re
                             sqlValue = new SqlDecimal((Decimal)value);
                         }
 
+                        SqlDecimal sqlValueBeforeAdjust = sqlValue;
```

Comment:
This should be moved into the Precision if statement below, since it isn't used anywhere else.